08_VirtualFileSystem/02_VirtualFileSystem.md

```diff
@@ -386,28 +397,29 @@ And we have the following code:
 
 ```c
 char *buffer[5];
-int sz = read(file_descriptor, buffer, 5);
-sz = read(file_descriptor, buffer, 5);
+int sz = read_resource(proc, file_handle, buffer, 5);
+sz = read_resource(proc, file_descriptor, file_handle, 5);
 ```
 
 The `buffer` content of the first read will be: `Text `, and the second one `examp`. This is the purpose  `buf_read_pos` variable in the file descriptor, so it basically needs to be incremented of nbytes, of course only if `buf_read_pos + nbytes < file_size` .
 The pseudocode for this function is going to be similar to the open/close:
 
 ```c
-ssize_t read(int fildes, void *buf, size_t nbytes) {
-    if (vfs_opened_files[fildes].fs_fildes_id != -1) {
-        int mountpoint_id = vfs_opened_files[fildes].mountpoint_id;
-        mountpoint_t *mountpoint = get_mountpoint_by_id(mountpoint_id);
-        int fs_file_id = vfs_opened_files[fildes].fs_file_id;
-        int bytes_read = mountpoints->operations->read(fs_file_id, buf, nbytes);
-        if (opened_files[fildes].buf_read_pos + nbytes < opened_files[fildes].file_size) {
-            opened_files[fildes].buf_read_pos += nbytes;
-        } else {
-            opened_files[fildes].buf_read_pos = opened_files[fildes].file_size;
-        }
-        return bytes_read;
-    }
-    return -1;
+ssize_t read(resource_t* res, void *buf, size_t nbytes) {
```

Comment:
Ok so this is read, and above we are using read_resource? And before we use again read. Let's decide on a name :D